app/models/foreman_kubevirt/kubevirt.rb

```diff
@@ -6,6 +6,7 @@ class Kubevirt < ComputeResource
     alias_attribute :token, :password
     alias_attribute :namespace, :user
     validates :hostname, :api_port, :namespace, :token, :presence => true
+    validate :test_connection
```

Comment:
will that attempt to connect to kubevirt each time an update is made to the compute resource ?
for which other cases this validation is invoked ?

Reply:
it will invoke when you do compute_resource.save

Reply:
so that makes sense to have.

But wouldn't we loose the ability to report what is the specific field that causes the issue ?
Meaning if we didn't specify hostname or api_port, the error might be different and rely on the response of test_connection.

Could you try to submit the form from the UI when any of the field is missing and see if we get the same notation of a missing attribute ?

Reply:
in the API, you get a full report:
2019-03-18T12:04:37 [D|app|0b2a6f65] Body: {
 |   "error": {"id":null,"errors":{"api_port":["can't be blank"],"base":["Failed to open TCP connection to ddd:443 (getaddrinfo: Name or service not known)"]},"full_messages":["Api port can't be blank","Failed to open TCP connection to ddd:443 (getaddrinfo: Name or service not known)"]}
 | }
 | 


from the UI:
![image](https://user-images.githubusercontent.com/5609929/54522591-4a1e9e80-4976-11e9-8951-b2849702efd0.png)